### PR TITLE
[Term Entry] PyTorch Tensor Operations .hstack()

### LINKED ISSUE #5468: [Term Entry] PyTorch Tensor Operations .hstack()
### Reason/inspiration (optional)

We would like a new entry on the `.hstack()` term under PyTorch. The entry should go in a new file under `docs/content/pytorch/concepts/tensor-operations/terms/hstack/hstack.md`.

The entry should include:

- An introduction to the concept
- A `Syntax` section that provides the syntax for the concept
- An `Example` section that provides an example demonstrating the concept in use

Please refer to the [term entry template](https://github.com/Codecademy/docs/blob/main/documentation/term-entry-template.md), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md) and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue.

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [ ] Issue is assigned

### PR DESCRIPTION
**Pull request by clonymontana**
### Description

-Added the  .hstack() function term under PyTorch
-Closes #5468 

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x ] My changes generate no new warnings.
- [x ] I have performed a self-review of my own writing and code.
- [x ] I have checked my entry and corrected any misspellings.
- [x ] I have made corresponding changes to the documentation if needed.
- [x ] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x ] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x ] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

File: content/pytorch/concepts/tensor-operations/terms/hstack/hstack.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  - 'Data Science'
+Tags:
+  - 'AI'
+  - 'Data Types'
+  - 'Deep Learning'
+  - 'Functions'
+CatalogContent:
+  - 'intro-to-py-torch-and-neural-networks'
+  - 'paths/data-science'
+---
```
Comment (mamtawardhani) [inline content/pytorch/concepts/tensor-operations/terms/hstack/hstack.md:15]:
```suggestion
---
Title: '.hstack()'
Description: 'Concatenates two or more tensors along the horizontal axis (column-wise)'
Subjects:
  - 'AI'
  - 'Data Science'
Tags:
  - 'AI'
  - 'Data Types'
  - 'Deep Learning'
  - 'Functions'
CatalogContent:
  - 'intro-to-py-torch-and-neural-networks'
  - 'paths/data-science'
---
```

Comment (mamtawardhani) [reply content/pytorch/concepts/tensor-operations/terms/hstack/hstack.md:15]: description should start with a verb

File: content/pytorch/concepts/tensor-operations/terms/hstack/hstack.md [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+In PyTorch, **`.hastack()`** (short for horizontal stack)is a function used to concatenate two or more tensors along the horizontal axis (axis1). This operation is useful for combining data with the same number of rows but differing in the number of columns. It acts similarly to numpy's 'np.hastack()' and is particulary handy when you're working with data that needs to be concatenated side by side before being fed into a model for training or inference.
```
Comment (mamtawardhani) [inline content/pytorch/concepts/tensor-operations/terms/hstack/hstack.md:17]:
```suggestion
In PyTorch, **`.hstack()`** (short for horizontal stack) is a function used to concatenate two or more tensors along the horizontal axis (`axis=1`). This operation is helpful in combining data with the same number of rows but differing in the number of columns. It acts similarly to NumPy's `np.hstack()` and is particularly handy for data that needs to be concatenated side by side before being fed into a model for training or inference.
```

Comment (mamtawardhani) [reply content/pytorch/concepts/tensor-operations/terms/hstack/hstack.md:17]: fixed spelling mistakes and reframed the sentence a bit

File: content/pytorch/concepts/tensor-operations/terms/hstack/hstack.md [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+
+# Syntax
+
+The basic syntax of `.hstack()` in PyTorch is as falows:
+
+```python
```
Comment (mamtawardhani) [inline content/pytorch/concepts/tensor-operations/terms/hstack/hstack.md:23]:
```suggestion
```pseudo
```

Comment (mamtawardhani) [reply content/pytorch/concepts/tensor-operations/terms/hstack/hstack.md:23]: syntax is to be wrapped in the ```pseudo``` block

File: content/pytorch/concepts/tensor-operations/terms/hstack/hstack.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+torch.hstack(tensors) -> Tensor
+```
+
+Where:
+
+ - `tensors` is a sequence of tesors with the same numbers of rows.
```
Comment (mamtawardhani) [inline content/pytorch/concepts/tensor-operations/terms/hstack/hstack.md:29]:
```suggestion
 - `tensors`: A sequence of tensors with the same number of rows. All tensors must have the same number of dimensions and the same size in all dimensions except for the dimension corresponding to the horizontal stacking.

Comment (mamtawardhani) [reply content/pytorch/concepts/tensor-operations/terms/hstack/hstack.md:29]: Can be elaborated

File: content/pytorch/concepts/tensor-operations/terms/hstack/hstack.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
...
+ - The function returns a new tensor containing the horizontal concatenation of the imput tensors.
```
Comment (mamtawardhani) [inline content/pytorch/concepts/tensor-operations/terms/hstack/hstack.md:30]:
```suggestion
The function returns a new tensor containing the horizontal concatenation of the input tensors.
```

Comment (mamtawardhani) [reply content/pytorch/concepts/tensor-operations/terms/hstack/hstack.md:30]: No bullet point needed as it is not a parameter

File: content/pytorch/concepts/tensor-operations/terms/hstack/hstack.md [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+
+```python
+import torch
+
+# Create two tensors
+a = torch.tensor([[1, 2],[3, 4]])
+b = torch.tensor([[5, 6],[7, 8]])
+
+# Horizontal stack
+c = toarch.hstack((a, b))
```
Comment (mamtawardhani) [inline content/pytorch/concepts/tensor-operations/terms/hstack/hstack.md:45]:
```suggestion
c = torch.hstack((a, b))
```

File: content/pytorch/concepts/tensor-operations/terms/hstack/hstack.md [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+
+Where:
+
+ - `tensors` is a sequence of tesors with the same numbers of rows.
+ - The function returns a new tensor containing the horizontal concatenation of the imput tensors.
+
+
+# Exemple
+
+Here's simple example demonstrating how '.hstack()' can be used to concatenate tensors:
```
Comment (mamtawardhani) [inline content/pytorch/concepts/tensor-operations/terms/hstack/hstack.md:35]:
```suggestion
Here's an example demonstrating how `.hstack()` can be used to concatenate tensors:
```

File: content/pytorch/concepts/tensor-operations/terms/hstack/hstack.md [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+# Horizontal stack
+c = torch.tensor((a, b))
+
+print(c)
+```
+In this Codebyte example, two 1-dimensional tensors are horizontally concatenated, showcasing how `.hstack()`seamlessly combines tensors not just in 2D but also in 1D tensors, further emphasizing the function's utility in handling tensors of varios dimensions as long as thay share the same number of rows (or are all 1-dimensional).
+
+
+
+
```
Comment (mamtawardhani) [inline content/pytorch/concepts/tensor-operations/terms/hstack/hstack.md:77]: This can be removed since the Codecademy Compiler does not have PyTorch dependencies yet.

**Review by mamtawardhani — CHANGES_REQUESTED** [review]
Hey @clonymontana thank you for contributing to Codecademy Docs 😄 

I've suggested some changes, could you please review and modify those at your earliest convenience? Thank you! 🚀

**Review by mamtawardhani — APPROVED** [review]
Thank you for contributing to Codecademy Docs @clonymontana 😄 

The entry looks good for a second round of review! 🚀

**Review by Sriparno08 — APPROVED** [review]
Looks good, @clonymontana!
